src/shared/utilities/timeoutUtils.ts

```diff
@@ -76,3 +76,26 @@ export class Timeout {
         }
     }
 }
+
+/**
+ * Invokes `fn()` until it returns a truthy value.
+ *
+ * @param opt.timeout  Timeout in ms (default: 5000)
+ * @param opt.interval  Interval in ms between fn() checks (default: 500)
+ * @param fn
+ *
+ * @returns Result of `fn()`, or `undefined` if timeout was reached.
+ */
+export async function waitUntil<T>(
```

Comment:
This doesn't really feel like a "waiting" function since we're executing `fn` every time the loops occurs, so we have the potential to return multiple times. If the goal is to execute one version of the promise and await that, could you use a `Promise.race` instead, racing the `fn` call against a longer-running promise?

Reply:
It's waiting for the result of fn() to change, as the docstring explains.


> If the goal is to execute one version of the promise and await that, could you use a `Promise.race` instead, racing the `fn` call against a longer-running promise?

Promise.race() doesn't work here (without an extra wrapper function) because the point is to wait until the result is not undefined, i.e. to provide the abstraction requested in https://github.com/aws/aws-toolkit-vscode/pull/1278#discussion_r491091073